client/js/editmode.js

```diff
@@ -1557,6 +1543,47 @@ async function updateWidget(currentState, oldState, applyChangesFromUI) {
   batchEnd();
 }
 
+async function updateWidgetId(widget, previousState) {
+  const children = Widget.prototype.children.call(widgets.get(previousState.id)); // use Widget.children even for holders so it doesn't filter
+  const cards = widgetFilter(w=>w.get('deck')==previousState.id);
+
+  for(const child of children)
+    sendPropertyUpdate(child.get('id'), 'parent', null);
+  for(const card of cards)
+    sendPropertyUpdate(card.get('id'), 'deck', null);
+  await removeWidgetLocal(previousState.id, true);
+  
+  const id = await addWidgetLocal(widget);
+
+  // Restore children
+  for(const child of children)
+    sendPropertyUpdate(child.get('id'), 'parent', id);
+  for(const card of cards)
+    sendPropertyUpdate(card.get('id'), 'deck', id);
+
+  // Change inheritFrom on widgets inheriting from altered widget
+  StateManaged.inheritFromMapping[id] = StateManaged.inheritFromMapping[previousState.id];
+  delete StateManaged.inheritFromMapping[previousState.id];
```

Comment:
Is this necessary? Those should be updated when receiving the delta..

Reply:
I removed them, but see issue #1540 .